core/src/main/java/com/linecorp/armeria/client/UserClient.java

```diff
@@ -158,7 +160,15 @@ protected final O execute(@Nullable EventLoop eventLoop,
         try (SafeCloseable ignored = ctx.push()) {
             return delegate().execute(ctx, req);
         } catch (Throwable cause) {
-            ctx.logBuilder().endResponse(cause);
+            final RequestLogBuilder logBuilder = ctx.logBuilder();
+            if (!ctx.log().isAvailable(RequestLogAvailability.REQUEST_START)) {
+                // An exception is raised even before sending a request, so end the request with the exception.
+                if (req instanceof HttpRequest) {
```

Comment:
nit: I'm not sure what's the best. How about moving these code to fallback of `DefaultHttpClient` and `DefaultTHttpClient`? Because I'm not sure that it's good choice that `HttpRequest` is used here.

Reply:
Agreed.

Reply:
Got it~!